presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -645,4 +661,25 @@ private static String getPartitionLocation(Table table, Optional<Partition> part
         }
         return partition.get().getStorage().getLocation();
     }
+
+    private static Optional<Domain> getPathDomain(TupleDomain<HiveColumnHandle> effectivePredicate)
+    {
+        if (!effectivePredicate.getColumnDomains().isPresent()) {
```

Comment:
Updated the code to use getDomains (will be reflected in the next iteration). The purpose is to check if "$path" has been passed as part of the predicate to the query and if that is the case then do the evaluation.

The new proposed code is:
```
private static Optional<Domain> getPathDomain(TupleDomain<HiveColumnHandle> effectivePredicate)
    {
        if (!effectivePredicate.getDomains().isPresent()) {
            return Optional.empty();
        }

        return effectivePredicate.getDomains().get().entrySet().stream()
                .filter(entrySet -> isPathColumnHandle(entrySet.getKey()))
                .findFirst()
                .map(entrySet -> entrySet.getValue());
    }
```